server/src/main/java/org/elasticsearch/action/bulk/BulkRequestParser.java

```diff
@@ -421,12 +421,22 @@ private static void checkBulkActionIsProperlyClosed(XContentParser parser) throw
         XContentParser.Token token;
         try {
             token = parser.nextToken();
-        } catch (XContentEOFException ignore) {
-            warnBulkActionNotProperlyClosed(
-                "A bulk action wasn't closed properly with the closing brace. Malformed objects are currently accepted but will be "
-                    + "rejected in a future version."
-            );
-            return;
+        } catch (XContentParseException e) {
+            Throwable cause = e.getCause();
+            if (cause != null) {
+                String exName = cause.getClass().getName();
```

Comment:
This workaround was necessary because we don't seem to have JsonEOFException on the classpath here. The class name is readable at runtime though it seems.

Reply:
You should not assume JsonEOFException exists, jackson is intentionally hidden, it is an implementation detail of xcontent.